components/Ripple/src/MDCRippleTouchController.m

```diff
@@ -30,13 +38,24 @@ - (instancetype)initWithView:(UIView *)view {
     _gestureRecognizer.delaysTouchesEnded = NO;
 
     _shouldProcessRippleWithScrollViewGestures = YES;
-    _view = view;
-    [_view addGestureRecognizer:_gestureRecognizer];
+    _rippleView = [[MDCRippleView alloc] init];
+  }
+  return self;
+}
 
-    _rippleView = [[MDCRippleView alloc] initWithFrame:view.bounds];
+- (void)addRippleToView:(UIView *)view {
+  if (_view) {
```

Comment:
This method feels similar to `addSubview:` in my mind, so the early-exit behavior here when a view's already been added to a hierarchy is surprising. I'd expect this call to always successfully add the ripple to the given view.

If you disagree, please provide more context and update the public API documentation to indicate that successive invocations of this method will do nothing or consider adding a failure signifier to the API.

Otherwise, please update this implementation to handle the case where the ripple is already part of an existing view.

Reply:
I don't feel strongly about the added check. It was more kept as a safety to block unaware clients to `initWithView:` and then `addRippleToView:` because they weren't aware of the documentation. I've removed the check as per your recommendation.

Reply:
A check of `if view == _view` seems fair, but the code you had in place would not have allowed a different view to be used after one had already been set.